### PR TITLE
Migrate to latest ESLint Version

### PR DESCRIPTION
**Pull request by dmanto**
### Update: Migrate to Latest ESLint Version

**Description:**

This PR introduces an upgrade to the latest version of ESLint. While there are multiple areas to review, I believe this is a solid starting point for improving code quality and consistency across the project.

**Summary of Changes:**

- **Commit 1:**
  - Separated Prettier from ESLint for better clarity and maintainability.
  - Migrated the configuration files to accommodate the latest ESLint version.
  
- **Commit 2:**
  - Resolved code issues based on the new ESLint rules.
  - _Note:_ Some rules may appear stricter than before. This is something we should evaluate and adjust as necessary.

**Next Steps:**

- Review the updated rules to ensure they align with our coding standards.
- Provide feedback on any rules that may need tweaking or further discussion.

**Comments and suggestions are welcome!**

### CODE REVIEW

File: eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                args: 'none',
+            },
+        ],
+        // Add more JS-specific rules here if needed
+    },
+}
+// Add the files for applying the recommended TypeScript configs 
+// only for the Typescript files.
+// This is necessary when we have the multiple extensions files 
+// (e.g. .ts, .tsx, .js, .cjs, .mjs, etc.).
```
Comment (kraih) [inline eslint.config.js:96]: Just remove the comments from the file, the object names are descriptive enough.

**Comment by kraih** [comment]
Guess that means we need a separate workflow to verify prettier formatting?

**Comment by dmanto** [comment]
> Guess that means we need a separate workflow to verify prettier formatting?

I can rename the existing prelint and prelint:fix scripts to format and format:fix to handle Prettier separately. Alternatively, I could keep the current names, which would avoid the need to run the two steps manually.

On second thought, prelint and prelint:fix could just call format and format:fix so we keep things DRY. Which approach do you prefer?

**Comment by kraih** [comment]
> > Guess that means we need a separate workflow to verify prettier formatting?
> 
> I can rename the existing prelint and prelint:fix scripts to format and format:fix to handle Prettier separately. Alternatively, I could keep the current names, which would avoid the need to run the two steps manually.
> 
> On second thought, prelint and prelint:fix could just call format and format:fix so we keep things DRY. Which approach do you prefer?

Afraid i have no idea what you're referring to. As far as i understand prettier used to be called as part of the eslint GitHub Action. Now that prettier is not in the eslint config anymore what will call it in CI?

**Comment by dmanto** [comment]
> > > Guess that means we need a separate workflow to verify prettier formatting?
> > 
> > 
> > I can rename the existing prelint and prelint:fix scripts to format and format:fix to handle Prettier separately. Alternatively, I could keep the current names, which would avoid the need to run the two steps manually.
> > On second thought, prelint and prelint:fix could just call format and format:fix so we keep things DRY. Which approach do you prefer?
> 
> Afraid i have no idea what you're referring to. As far as i understand prettier used to be called as part of the eslint GitHub Action. Now that prettier is not in the eslint config anymore what will call it in CI?

Prettier will still run as part of the CI process because I added pre scripts for both lint and lint:fix (prelint and prelint:fix). npm automatically runs these pre scripts before executing the corresponding lint and lint:fix scripts. So, Prettier will be invoked every time linting occurs, both in local development and CI.

**Pull request merged**